magellan-library/src/main/java/com/wealthfront/magellan/navigation/LazySetNavigator.kt

```diff
@@ -129,5 +129,7 @@ public open class LazySetNavigator(
     }
   }
 
-  override fun onBackPressed(): Boolean = currentNavigable?.backPressed() ?: false
+  override fun backPressed(): Boolean {
```

Comment:
either we override LazySetNavigator.backPressed (instead of the usual onBackPressed), or we change LifecycleRegistry.onAllActiveListenersUntilTrue to filter out NO_LIMIT instead of SHOWN. I chose this option for being lower-impact